lib/logstash/outputs/s3.rb

```diff
@@ -106,6 +109,13 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   # S3 bucket
   config :bucket, :validate => :string, :required => true
 
+  # Specify a custom endpoint for use with non-AWS S3 implementations, e.g.,
+  # Ceph.  Provide a URL in the format http://127.0.0.1:8080/
+  config :endpoint, :validate => :string
```

Comment:
* Endpoint is the name in the AWS documentation, but it's NOT a URL, it's a host and port.
* Eg `boto.s3.S3Connection` has a `host` parameter that takes an optional port, eg `127.0.0.1:8080`.